library/core/src/panic/panic_info.rs

```diff
@@ -46,7 +66,23 @@ impl<'a> PanicInfo<'a> {
         location: &'a Location<'a>,
     ) -> Self {
         struct NoPayload;
-        PanicInfo { location, message, payload: &NoPayload }
+        PanicInfo { location, message, payload: PayloadKind::Any(&NoPayload) }
+    }
+
+    #[unstable(
+        feature = "panic_internals",
+        reason = "internal details of the implementation of the `panic!` and related macros",
+        issue = "none"
+    )]
+    #[doc(hidden)]
+    #[inline]
+    #[cfg(not(bootstrap))]
+    pub fn error_constructor(
+        // message: Option<&'a fmt::Arguments<'a>>,
```

Comment:
This will likely need a message as well, so that `unwrap` and `expect` can provide their message when panicking as an error. My plan is to change the `error()` method on `PanicInfo` to be called `source`, and to treat the message like the outermost error in the chain of errors.